presto-orc/src/main/java/com/facebook/presto/orc/reader/LongDictionaryStreamReader.java

```diff
@@ -84,7 +86,7 @@ public LongDictionaryStreamReader(Type type, StreamDescriptor streamDescriptor,
             throws OrcCorruptionException
     {
         requireNonNull(type, "type is null");
-        verifyStreamType(streamDescriptor, type, t -> t instanceof BigintType || t instanceof IntegerType || t instanceof SmallintType || t instanceof DateType);
+        verifyStreamType(streamDescriptor, type, t -> t instanceof BigintType || t instanceof IntegerType || t instanceof SmallintType || t instanceof DateType || t instanceof TimeType || t instanceof TimestampType);
```

Comment:
We should keep the original checking rather than bypassing it only for Raptor. It's ok to have some hacky workaround in `presto-raptor` than `presto-orc`. Let's chat with @jessesleeping to see how to solve this issue.

Reply:
We will need to convert the actual column type to "storage type" (see `StorageTypeConverter`) in wherever Raptor calls `OrcReader::createRecordreader()`:
- `ShardStats`
- `OrcStorageManager`
- `OrcPagefileRewriter`